common/persistence/persistence_rate_limited_clients.go

```diff
@@ -55,54 +55,66 @@ var (
 		Scope:   enumspb.RESOURCE_EXHAUSTED_SCOPE_NAMESPACE,
 		Message: "Namespace Persistence Max QPS Reached.",
 	}
+	ErrPersistenceNamespaceShardLimitExceeded = &serviceerror.ResourceExhausted{
+		Cause:   enumspb.RESOURCE_EXHAUSTED_CAUSE_PERSISTENCE_LIMIT,
+		Scope:   enumspb.RESOURCE_EXHAUSTED_SCOPE_NAMESPACE,
+		Message: "Namespace Per-Shard Persistence Max QPS Reached.",
```

Comment:
Is this error message exposed to users? Shards are an internal concept, so I think we should mask that with a more generic error message and just log the more descriptive string (with shard ID might be helpful, if it's available).

Reply:
We return this message to SDK. I quickly checked sdk-go. Looks like it does not do anything with the error message. It records some metrics based on the Cause field. 
Also, we don't print this error message here. The caller logs it. Shard id is also logged there.